### ISSUE TITLE
HVACVariableRefrigerantFlow divide by zero in CalcVRFCondenser_FluidTCtrl

### ISSUE THREAD

**Issue by DeadParrot** [NREL/EnergyPlus#5374]
Test case VariableRefrigerantFlow_FluidTCtrl_5Zone run with a GCC debug build gives a core dump due to an arithmetic exception that is almost certainly a divide by zero in CalcVRFCondenser_FluidTCtrl at [this line](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L7922) (last line in this block):

``` cpp
Pipe_v_ref = Pipe_m_ref / ( 3.141593 * pow_2( VRF( VRFCond ).RefPipDia ) * 0.25 ) / GetSupHeatDensityRefrig( VRF( VRFCond ).RefrigerantName, Pipe_T_IU_in, max( min( Pcond, RefPHigh ), RefPLow ), RefrigerantIndex, RoutineName );
Pipe_Num_Re = Pipe_m_ref / ( 3.141593 * pow_2( VRF( VRFCond ).RefPipDia ) * 0.25 ) * VRF( VRFCond ).RefPipDia / Pipe_viscosity_ref * 1000000;
Pipe_Num_Pr = Pipe_viscosity_ref * Pipe_cp_ref * 0.001 / Pipe_conductivity_ref;
Pipe_Num_Nu = 0.023 * std::pow( Pipe_Num_Re, 0.8) * std::pow( Pipe_Num_Pr, 0.4);
Pipe_Num_St = Pipe_Num_Nu/Pipe_Num_Re/Pipe_Num_Pr;
```

Note that there is a second occurrence of the same statement in the file.

**Comment by mjwitte** [comment]
And this didn't fail in CI?  Did you force design days only (as CI does) or run the file as-is (both design days and short weather file runperiods)?

**Comment by mjwitte** [comment]
Oh - it _is_ failing on CI [here](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-9ff9bbb09e88e79f373beb6a1e852aafe19c857d-x86_64-Linux-Ubuntu-14.04-gcc-4.8-IntegrationCoverage-Debug.html#test)

**Comment by amirroth** [comment]
Stuart,

Is it possible that you are accidentally making these changes in the
HBIRE_vector_take2 branch?

-Amir

**Comment by DeadParrot** [comment]
@amirroth No, this is running straight develop code and it is not related to anything I'm changing. I just hit this and wanted to report it.

@mjwitte I'm running all the example files as is as a baseline for comparing develop with revisions I'm working on (in this case the next ObjexxFCL release). So it is running whatever period the idf specifies.

**Comment by Myoldmopar** [comment]
Well.  Sh....enanigans.  OK.  @rongpengzhang @rraustad Can one of you offer a fix to this this morning so we can put out the bug fix release without this issue?  If not, I'll start reverting things back out.......

**Comment by amirroth** [comment]
@deadparrot, @mjwitte, @myoldmopar. That is very strange, because someone
is doing some VRF related work in this branch, and for everyone's sake it
better not be me.

Macintosh-2:EnergyPlusTeam amir$ git status
On branch HBIRE_vector_take2
Your branch is up-to-date with 'origin/HBIRE_vector_take2'.
You have unmerged paths.
  (fix conflicts and run "git commit")

Changes to be committed:

```
modified:   src/EnergyPlus/DXCoils.cc
modified:   src/EnergyPlus/DXCoils.hh
modified:   src/EnergyPlus/HVACVariableRefrigerantFlow.cc
modified:   src/EnergyPlus/HVACVariableRefrigerantFlow.hh
modified:   testfiles/VariableRefrigerantFlow_FluidTCtrl_5Zone.idf
```

Unmerged paths:
  (use "git add <file>..." to mark resolution)

```
both modified:   tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc
```

**Comment by kbenne** [comment]
`git diff --staged` to see what you are about to commit. `git reset .` to unstage. I bet those files diff only by line endings.

**Comment by DeadParrot** [comment]
@amirroth GitHub "Blame" attributes the code to @rongpengzhang and it is from Jul 19.

Minor aside: the use of a low precision Pi literal instead of the value from DataGlobals should have been caught on review.

**Comment by kbenne** [comment]
Also I didn't notice when I originally commented that you are in the middle of conflict resolution. Maybe you should do `git merge --abort` until you figure out what is going on.

**Comment by rraustad** [comment]
Pipe_m_ref = 0 at the beginning of this function and might also be 0 during the calculation of Pipe_Num_Re. Not sure I have sufficient knowledge of the model to make a change. There is a SafeDiv function that might benefit multiple equations in the module. @rongpengzhang should review this issue.

**Comment by Myoldmopar** [comment]
@rongpengzhang I need you to comment back on whether you can provide a fix for this very soon or I'll revert out #5367.

**Comment by rraustad** [comment]
Why are any of these calculations being performed if refrigerant mass flow = 0?

**Comment by rongpengzhang** [comment]
@Myoldmopar  This should be a minor problem. I will fix it this morning.

**Comment by rongpengzhang** [comment]
@Myoldmopar @mjwitte @DeadParrot I cannot reproduce the problem on my machine (Windows). Any ideas on how to reproduce it?

Here is the build message, without any warnings:
![image](https://cloud.githubusercontent.com/assets/7785946/11848754/08e988c6-a3d9-11e5-9e10-f832db7b115d.png)

Here is the message running the example files (VariableRefrigerantFlow_FluidTCtrl_5Zone.idf):
![image](https://cloud.githubusercontent.com/assets/7785946/11848774/2f609418-a3d9-11e5-9ac3-6c0a4450243e.png)

But anyway, I think @rraustad 's suggestion will make it more robust and solve the problem.

**Comment by Myoldmopar** [comment]
@rongpengzhang I guess I wouldn't be surprised if the divide-by-zero was being muffled by the MSVC compiler settings.  I don't have that set up right in front of me or I would verify it for you.  @mjwitte or @rraustad Can you confirm the behavior on your machines?  We'll chat about this on the iteration call.

**Comment by mjwitte** [comment]
Win64 debug build does not fail for me.  Divide by zeros _used_ to throw an exception in debug builds.  But @DeadParrot said "almost certainly a divide by zero".

**Comment by rraustad** [comment]
@rongpengzhang also check the headings in the html report. Design Size Rated Sensible Heat Ratio [m3/s] and Design Size Indoor Unit Reference Subcooling [m3/s] should be Design Size Maximum Flow Rate [m3/s].

![vrf_newmodelhtml](https://cloud.githubusercontent.com/assets/5778156/11849823/da5ddb14-a3f7-11e5-85bd-507c7157bd48.png)

**Comment by rongpengzhang** [comment]
Thanks @rraustad for the suggestion.

**Comment by rongpengzhang** [comment]
@Myoldmopar @mjwitte @DeadParrot I just finish the modification. I cannot reproduce the problem, so I'm not sure whether this is really the solution or not. Let's see.

**Comment by rongpengzhang** [comment]
The new commit goes in #5378

**Comment by DeadParrot** [comment]
@rongpengzhang this is a floating point divide by zero and to get those caught with VC builds I think we'd need to add floating point handler setup code like we did for Linux.

**Comment by mjwitte** [comment]
Closed via #5378
